hydra-web/src/js/app.js

```diff
@@ -180,25 +199,66 @@ function(
             $("#loginToken")[0].value = "token";
             app.loginDialog = alertify.minimalDialog($('#loginForm')[0]);
         },
+        unsudo: function(username, token) {
+            var unsudoUrl = "/authentication/unsudo";
+            $.ajax({
+                type: 'POST',
+                url: unsudoUrl,
+                data: {
+                    username: username,
+                    token: token,
+                    sudo: Cookies.get("sudo"),
+                },
+            });
+            Cookies.set("sudo", "", {expires:0});
+            app.user.set("sudo", "");
+            app.sudoCb(false, true);
+        },
+
         sudo:function() {
+            var sudoUrl = app.authprefix() + "/authentication/sudo";
```

Comment:
Consider refactor this function - I feel the way it mixes username/token validation and  `unsudo` makes it less readable than it can be.